stagemonitor-requestmonitor/src/main/java/org/stagemonitor/requestmonitor/MonitoredMethodRequest.java

```diff
@@ -49,6 +51,16 @@ public Object execute() throws Exception {
 	}
 
 	@Override
+	public ListenableFuture<Object> executeAsync() {
+		try {
+			return Futures.immediateFuture(methodExecution.execute());
+		}
+		catch (Exception e) {
+			return Futures.immediateFailedFuture(e);
```

Comment:
No, it returns a Future that re-throws the exception when it completes (a call to get()), or calls the registered listeners with the exception.